lix/packages/sdk/src/combine/combine-changes.ts

```diff
@@ -0,0 +1,106 @@
+import type { LixPlugin } from "../plugin.js";
+import type { Change } from "../schema.js";
+import type { Lix } from "../types.js";
+
+/**
+ * Combines the changes of two lixes into the target lix.
+ */
+export async function combineChanges(args: { target: Lix; source: Lix }): Promise<void> {
+	const toBeAppliedChanges: Record<LixPlugin["key"], Array<Change>> = {};
+
+	// TODO ideally a right join via sqlite attaching works
+	//      instead of fetching all changes from source
+	const sourceChanges = await args.source.db
+		.selectFrom("change")
+		.selectAll()
+		.execute();
+
+	// 1. Get the changes that exists in source but not in target
+	for (const change of sourceChanges) {
+		const targetChange = await args.target.db
+			.selectFrom("change")
+			.select("id")
+			.where("id", "=", change.id)
+			.executeTakeFirst();
+
+		// change already exists, skip
+		if (targetChange !== undefined) {
+			continue;
+		}
+		// copy change to target
+		await args.target.db
```

Comment:
this will lead to missing commits from source db in target db.

Reply:
also this will only work with confirmed branch support in the existing implementation so all changes not referenced by the current commit and change lineage are ignored.

Reply:
copying the commits over has been implemented